indico/modules/events/registration/client/js/form/fields/DateInput.jsx

```diff
@@ -8,69 +8,120 @@
 import createDecorator from 'final-form-calculate';
 import PropTypes from 'prop-types';
 import TimePicker from 'rc-time-picker';
-import React, {useState} from 'react';
+import React from 'react';
 import {Field} from 'react-final-form';
 import {Form} from 'semantic-ui-react';
 
 import {SingleDatePicker} from 'indico/react/components';
-import {FinalDropdown, parsers as p} from 'indico/react/forms';
+import {FinalDropdown, FinalField, FinalInput, parsers as p} from 'indico/react/forms';
 import {Translate} from 'indico/react/i18n';
+import {toMoment} from 'indico/utils/date';
 
 import '../../../styles/regform.module.scss';
 
-export default function DateInput({htmlName, disabled, id, dateFormat, timeFormat}) {
-  const [date, setDate] = useState(null);
-  const [time, setTime] = useState(null);
+function DateInputComponent({value, onChange, disabled, required, dateFormat, timeFormat}) {
+  const dateValue = value.includes(' ') ? value.split(' ')[0] : value;
+  const timeValue = value.includes(' ') ? value.split(/ (.*)/)[1] : '';
+  const timeMomentFormat = timeFormat === '12h' ? 'hh:mm a' : 'HH:mm';
+  const handleDateChange = newDate =>
+    onChange(
+      timeFormat ? `${newDate.format(dateFormat)} ${timeValue}` : newDate.format(dateFormat)
```

Comment:
The value shouldn't be formatted using the selected format but rather use iso format (`YYYY-MM-DD HH:MM` or whatever is present - so `YYYY` or `YYYY-MM` for those formats, `YYYY-MM-DD` for all standard dates and the first one for anything with a date+time, and `HH:MM` if it's just the time)

Reply:
I'm using the selected format because [it's what the `DateField` is expecting](https://github.com/indico/indico/blob/665192dee2746a615018480b1334bba6d74d7a69/indico/modules/events/registration/fields/simple.py#L175). But I agree, we should be using ISO. I will take care of that then

Reply:
Unless you already changed it, let's make this a separate task then

Reply:
ah nevermind, already updated. nice! :)